File: sujeong/week4/큰 수 만들기.swift
import Foundation

func solution(_ number:String, _ k:Int) -> String {
    var answer = ""

    var _k:Int = k
    let numbers:[Character] = Array(number)
    let n:Int = numbers.count
    var stack:[Character] = []
    
    for (i,num) in numbers.enumerated() {
        while _k > 0
                && !stack.isEmpty
                && stack.last!.wholeNumberValue! < num.wholeNumberValue! {
            stack.removeLast()
            _k -= 1
        }
        if stack.count < n-k {
            stack.append(num)
        }
    }
    return String(stack)
}

File: chaewon/week4/큰수만들기.py
def solution(number, k):
    answer = []

    for num in number:
        while len(answer) > 0 and answer[-1] < num and k > 0:
            k -= 1
            answer.pop()
        answer.append(num)
    answer = answer[:-k] if k != 0 else answer

    return ''.join(answer)